RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -28,108 +25,61 @@ public Response(int responseCode, String responseBody, String responseMessage) {
     }
 
     public final ConnectionDetails connectionDetails;
+    private final HttpClient client;
 
     public RestClient(ConnectionDetails connectionDetails) {
         this.connectionDetails = connectionDetails;
-    }
-
-    public Response get(String path, boolean quietLogging) throws Exception {
-        String urlString = connectionDetails.url + "/" + path;
-        
-        URL url = new URL(urlString);
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-
-        // Set the request method
-        conn.setRequestMethod("GET");
-        
-        // Set the necessary headers
-        setAuthHeader(conn);
-
-        // Enable input and output streams
-        conn.setDoOutput(true);
-
-        // Report the results
-        int responseCode = conn.getResponseCode();
-
-        String responseBody;
-        if (responseCode >= HttpURLConnection.HTTP_BAD_REQUEST) {
-            // Read error stream if present
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getErrorStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        } else {
-            // Read input stream for successful requests
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        }
-
-        if (quietLogging || (responseCode == HttpURLConnection.HTTP_CREATED || responseCode == HttpURLConnection.HTTP_OK)) {
-            logger.debug("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        } else {
-            logger.error("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        }
-
-        conn.disconnect();
 
-        return new Response(responseCode, responseBody, conn.getResponseMessage());
+        this.client = HttpClient.create()
+            .baseUrl(connectionDetails.url)
+            .headers(h -> {
+                h.add("Content-Type", "application/json");
```

Comment:
Is this always true for ES/OS?  I feel like I've seen some whacky content types, including PNGs and fonts (no idea why) go in as documents.  Would it make sense to try to run those as batch puts?  I wouldn't worry about it much - just put a check somewhere on the doctypes and throw if it's something truly whacky.

Reply:
I honestly don't even know how to check for that at this point; I'll make a JIRA to help capture the miss here.

Reply:
Done - https://opensearch.atlassian.net/browse/MIGRATIONS-1718